src/Carnac.Logic/KeyProvider.cs

```diff
@@ -101,10 +106,19 @@ KeyPress ToCarnacKeyPress(InterceptKeyEventArgs interceptKeyEventArgs)
                            interceptKeyEventArgs.Key <= Keys.Z;
 
             var inputs = ToInputs(isLetter, winKeyPressed, interceptKeyEventArgs);
+            try
```

Comment:
Yeah it would be better not to pull the related icon if it's not needed, but I saw you don't have access to the application's settings from this class, and I didn't want to reorganize it at this point. Let me know if you think it's a good idea to supply the settings to this class (like you did in MessageProvider class the gets settings in the ctor)